File: app/src/main/java/com/omar/retromp3recorder/app/recording/usecase/ChangeBitrateUC.java
package com.omar.retromp3recorder.app.recording.usecase;

import com.omar.retromp3recorder.app.recording.recorder.VoiceRecorder;
import com.omar.retromp3recorder.app.recording.repo.BitRateRepo;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ChangeBitrateUC {
    private final BitRateRepo repo;

    @Inject
    public ChangeBitrateUC(BitRateRepo repo) {
        this.repo = repo;
    }

    public Completable execute(VoiceRecorder.BitRate bitRate) {
        return Completable.fromAction(() -> repo.newValue(bitRate));
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/recording/usecase/ChangeBitrateUC.kt
package com.omar.retromp3recorder.app.recording.usecase

import com.omar.retromp3recorder.app.recording.recorder.VoiceRecorder.BitRate
import com.omar.retromp3recorder.app.recording.repo.BitRateRepo
import io.reactivex.Completable
import javax.inject.Inject

class ChangeBitrateUC @Inject constructor(private val repo: BitRateRepo) {
    fun execute(bitRate: BitRate): Completable {
        return Completable.fromAction { repo.newValue(bitRate) }
    }
}
